timesheets/Code.gs

```diff
@@ -0,0 +1,149 @@
+  rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues).build();
+  bigRange.setDataValidation(rule);
+  bigRange.setValue('NOT NOTIFIED');
+}
+
+/**
+* Runs the "Calculate Pay" script, creating a new column with each employee's 
+* pay based on their reported hours and hourly wage.
+*/
+function calculatePay() {
+  // declares variables
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastRow = sheet.getLastRow();
+  var lastCol = sheet.getLastColumn();
+  var frozenRows = sheet.getFrozenRows();
+ 
+  // gets hours & hourly rate values
+  var rangeValues = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.HOURS_START, lastRow - frozenRows, 6).getValues();
+  var pays = [];
+  
+  // goes through every employee
+  for (var i = 0; i <= lastRow - (frozenRows + 1); i++) {
+    var numHours = 0;
+    for (var j = 0; j < 5; j++) {
+      numHours += rangeValues[i][j];
+    
+    // pushes weekly pay value into 2D array
+    pays.push([numHours * sheet.getRange(i + (frozenRows + 1), COLUMN_NUMBER.HOURLY_PAY).getValue()]);
+  
+  // pushes entire 2D array of pay values into column 
+  var pushRange = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.CALC_PAY, lastRow - 1, 1).setValues(pays);
+}
+
+/**
+* Runs the script "Send Emails". Sends e mails to every employee notifying 
+* them whether or not their timesheet has been approved.
+* Checks + updates "NOTIFIED" status so employers do not send e mails more than once.
+*/
+function sendEmails() {
+    var sheet = SpreadsheetApp.getActiveSheet();
+    var lastRow = sheet.getLastRow();
+    var lastCol = sheet.getLastColumn();
+    // lastCol here is the NOTIFIED column
+    var frozenRows = sheet.getFrozenRows();
+  
+    // gets ranges of values all at once
+    var emailValues = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.EMAIL, lastRow - frozenRows, 1).getValues();
+    var approvalValues = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.APPROVAL, lastRow - frozenRows, 1).getValues();
+    var notifiedValues = sheet.getRange(frozenRows + 1, lastCol, lastRow - frozenRows, 1).getValues();
+  
+    // goes through the whole email column
+    for (var i = 0; i <= lastRow - (frozenRows + 1); i++) { 
+      // don't notify twice
+      if (notifiedValues[i][0] == 'NOTIFIED') {
+        continue;
+      
+           
+      var email = emailValues[i][0];
+      var approvalValue = approvalValues[i][0];    
+      
+      // approval email
+      if (approvalValue == 'APPROVED') {
+        var message = 'Your timesheet has been approved';
+        var subject = 'TimeSheet Approval';
+      } else if (approvalValue == 'NOT APPROVED') {
+        var message = 'NOT APPROVED';
+        var subject = 'TimeSheet not Approved';
+      } else if (approvalValue == 'IN PROGRESS') {
+        continue;
+      } 
+      MailApp.sendEmail(email, subject, message);
+      notifiedValues[i][0] = 'NOTIFIED';
+      var notifiedUpdate = sheet.getRange(i, lastCol).setValue('NOTIFIED');
```

Comment:
`setValue` doesn't return anything interesting, no reason to have a variable here.